src/main/kotlin/dev/znci/twine/TwineNative.kt

```diff
@@ -151,6 +177,64 @@ abstract class TwineNative(
             }
         }
     }
+
+    /**
+     * Converts Lua varargs to Kotlin vararg arguments based on the parameter type.
+     *
+     * @param param The parameter whose type will be used for conversion.
+     * @return An array of Kotlin compatible vararg arguments.
+     */
+    private fun Varargs.toKotlinVarargArgs(param: KParameter): Array<Any?> {
```

Comment:
I don't like how we have two functionally similar type converters now. Could we not create an `Array<Any>` when we need it and just use our normal type converter?